packages/core/src/domain/patient.ts

```diff
@@ -90,3 +91,44 @@ export interface Patient extends BaseDomain, PatientCreate {}
 export function getStatesFromAddresses(patient: Patient): USState[] {
   return patient.data.address.map(getState);
 }
+
+export function patientDataFromResource(patientResource: PatientResource): PatientData | undefined {
+  const firstName = patientResource.name[0]?.given[0];
+  const lastName = patientResource.name[0]?.family;
+  const dob = patientResource.birthDate;
+  const genderAtBirth = mapGender(patientResource.gender);
+  const addresses = getPatientAddresses(patientResource);
+
+  if (!firstName || !lastName || !dob || !genderAtBirth) return;
+  if (!addresses.length) return;
```

Comment:
If there's no address we don't want to store/use what we have?

And I'm assuming the other fields are required?

Reply:
Yea, all those fields are required for patient matching